Microsoft.Toolkit.Uwp.Input.GazeControls/GazeKeyboard.xaml

```diff
@@ -0,0 +1,21 @@
+﻿<UserControl
```

Comment:
This should be a templated control over a UserControl. I think it'd make sense to potentially just have each type of keyboard be it's own style. Need to think about the loading them dynamically, it's a bit odd for a control library.

I think given that the three layouts you have probably cover most scenarios, it should be just as easy for someone to provide their own style to layout a different keyboard as the existing setup is.

Reply:
Can  you please elaborate what you mean by "This should be a templated control over a UserControl."?

Dynamic loading is a highly desired capability. The three layouts show three categories, but there is a large variation in user needs to be able to customize a soft keyboard for their particular needs. When working with speech language pathologists, this is requested quite often. What they would ideally like is a visual editor where they can work with a particular user and design and layout specific to that user's needs at their stage of illness. These basic capabilities are a starting point. Hopefully we will be able to get the advanced scenarios.

Reply:
You can still have different styles and dynamics, but you get that from making it a TemplatedControl. That gives the developer using it complete control to modify how the control looks on top of its behavior. You can read more in the first couple of sections of the [XAML Styles](https://docs.microsoft.com/en-us/windows/apps/design/style/xaml-styles) docs and [control templates](https://docs.microsoft.com/en-us/windows/apps/design/style/xaml-control-templates).

There'd be a default layout which would be the base style, but then the developer could apply the alternate styles provided by you or create their own for the other layouts.

Reply:
I reviewed control templates. It appears that they allow me to change the look and feel of the control by using the Template property of the control. My control already provides that and much more right now. All my user control has right now is a blank grid that acts as the layout root and the entire user's template gets loaded there. So the user is already free to completely customize the look and behavior by simply specifying the LayoutUri property.

I also looked into refactoring my code to use the Template property instead of my LayoutUri property. I am able to successfully load my layout from file as a ControlTemplate. But when I try to assign it to the Template property, I get this error:
```
WinRT information: Setting the Template property on a UserControl is not supported.
```

Reply:
Creating Templated Controls is a different process than a UserControl, it's not something you add to the UserControl. You need to make your XAML a resource dictionary with a Style, then your code-behind uses an override to OnApplyTemplate to grab specific elements within the style to hook events into. This means that a user can provide a custom style to the control without having to worry about it's behavior as long as they have the core elements.

This was the main thing I was trying to say how each of your keyboard layouts become an alternate style (like how we have for the `InAppNotification` control/sample). We want to avoid reading in loose XAML from a StorageFile for a control library.

Your code should be 80-90% the same, but just structured from a different perspective.